src/ORM/Filters/UsesSearchFilters.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace SilverStripe\ORM\Filters;
+
+use SilverStripe\Core\Injector\Injector;
+
+trait UsesSearchFilters
+{
+    /**
+     * Given a filter expression and value construct a {@see SearchFilter} instance
+     *
+     * @param string $filter E.g. `Name:ExactMatch:not`, `Name:ExactMatch`, `Name:not`, `Name`
+     * @param mixed $value Value of the filter
+     * @return SearchFilter
+     */
+    protected function createSearchFilter($filter, $value)
+    {
+        // Field name is always the first component
+        $fieldArgs = explode(':', $filter);
+        $fieldName = array_shift($fieldArgs);
+        $default = 'DataListFilter.default';
+
+        // Inspect type of second argument to determine context
+        $secondArg = array_shift($fieldArgs);
```

Comment:
Don't use `array_shift()` here, it's not obvious that it returns null if there's no second arg (I have to check this locally with php -a).  Just do an `if (count($fieldArgs) > 0) { $secondArg = $fieldArgs[1]; } else { $secondArg = ''; }` or similar

Reply:
As mentioned in https://github.com/silverstripe/silverstripe-framework/pull/10925#discussion_r1306883875 this is just a straight cut & paste from `DataList` - I'm not inclined to change the implementation of this all of a sudden. That's out of scope for this PR.